### PR TITLE
Feature: adding order repository mapping with orm

### PR DESCRIPTION
**Pull request by eapg**
* Adding order repository mapping with orm
* Adding order fixture to match with orm
* Adding order repository test

### CODE REVIEW

File: src/lib/repositories_v2/impl/order_repository_impl.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            .filter(OrderDetail.order_id.in_([order.id for order in orders_to_process]))
+            .group_by(OrderDetail.order_id)
+            .all()
+        )
+        validation_order_map = dict(
+            (product_id, result) for product_id, result in validation_orders
+        )
+        return validation_order_map
+
+    def reduce_order_ingredients_from_inventory(self, order_id):
```
Comment (reynoldmorel) [inline src/lib/repositories_v2/impl/order_repository_impl.py:129]: You can do this using cte

File: src/lib/repositories_v2/order_repository.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Interface for the repositories through Abstract method
+from abc import ABCMeta
+
+from src.lib.repositories.generic_repository import GenericRepository
+
+
+class OrderRepository(GenericRepository, metaclass=ABCMeta):
```
Comment (reynoldmorel) [inline src/lib/repositories_v2/order_repository.py:7]: You don't need to copy and paste this if it already exists, the same for the other existing repositories

File: src/lib/repositories_v2/impl/order_repository_impl.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            .join(
+                InventoryIngredient,
+                ProductIngredient.ingredient_id == InventoryIngredient.ingredient_id,
+            )
+            .filter(
+                OrderDetail.entity_status == Status.ACTIVE.value,
+                ProductIngredient.entity_status == Status.ACTIVE.value,
+                InventoryIngredient.entity_status == Status.ACTIVE.value,
+            )
+            .filter(OrderDetail.order_id.in_([order.id for order in orders_to_process]))
```
Comment (reynoldmorel) [inline src/lib/repositories_v2/impl/order_repository_impl.py:120]: Is confusing to have a mix of `filter` and `,` to perform `and` conditions, could do it one way if possible?

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/lib/repositories_v2/order_repository.py (hunk 2 above)
Comment (eapg) [reply src/lib/repositories_v2/order_repository.py:7]: i create a new one in the cases where i dont have the same methods(functions) as version v1 of the project.

File: src/lib/repositories_v2/impl/order_repository_impl.py (hunk 3 above)
Comment (eapg) [reply src/lib/repositories_v2/impl/order_repository_impl.py:120]: Done.

File: src/lib/repositories_v2/order_repository.py (hunk 2 above)
Comment (eapg) [reply src/lib/repositories_v2/order_repository.py:7]: This part will be fixed in a refactor PR.

**Review by reynoldmorel — APPROVED** [review]
Nit: format the query and create a variable for this query outside of the repository class.

**Pull request merged**